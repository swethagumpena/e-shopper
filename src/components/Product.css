.product{
    border: 0.125em solid black;
    padding: 0.5em;
    margin: 1em;
}

.product img{
    height: auto;
    max-height: 250px;
    width: auto;
    max-width: 250px;
}

.container {
    font-family:'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    font-size: large;
    padding: 0.5em;
}

