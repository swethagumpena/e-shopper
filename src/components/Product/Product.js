import React from 'react';
import PropTypes from 'prop-types';
import Counter from '../Counter/Counter';
import styles from './Product.module.scss';

const Product = ({ product, onIncrement, onDecrement }) => (
  <div className={styles.productCard}>
    <img data-testid="product-img" id={styles.productImg} src={product.url} alt="product" />
    <div className={styles.priceQuantityContainer}>
      <h4>{product.productName}</h4>
      <p>1 kg</p>
      <div className={styles.priceQuantity}>
        <p className={styles.productPrice}>
          {`MRP ${product.price}/-`}
        </p>
        <Counter
          value={product.quantity}
          onIncrement={onIncrement}
          onDecrement={onDecrement}
        />
      </div>
    </div>
  </div>
);

const productShape = {
  id: PropTypes.number,
  productName: PropTypes.string,
  quantity: PropTypes.number,
  price: PropTypes.number,
  url: PropTypes.string,
};

Product.propTypes = {
  // product -> object
  product: PropTypes.shape(productShape).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default Product;
