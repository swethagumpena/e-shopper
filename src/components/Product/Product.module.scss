@import '../../styles/index.scss';

.categories{
    // display: flex;
}

.productCard{
    border: 0.125rem solid lightgray;
    padding: 0.5rem;
    margin: 0.5rem;
    box-shadow: 0rem 0rem 0.25rem #9a9a9a;
    border-radius: 0.25rem;
    width: 15.5rem;
}

#productImg {
    width: 100%;
    height: 15rem;
}

.priceQuantityContainer {
    font-family:'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    font-size: large;
    padding: 0.5rem;
}

.priceQuantity{
    display: flex;
    flex-direction: column;
    background-color: #d0d0d0;
    p {
        padding: 0 1rem
    }
}

// .priceQuantity p {
//     padding: 0 1rem;
// }

.productPrice{
    font-weight: bold;
}

@include media-mixin($laptop-breakpoint){
    .productCard {
        width:  calc((100% - (6 * 0.5rem + 6 * 0.5rem + 6 * 0.125rem)) / 3);
    }
    #productImg {
        height: 18.75rem;
    }
}
// @media (max-width: 72rem) {         /*ipad pro/laptop: 1024*/
//     .productCard {
//         width:  calc((100% - (6 * 0.5rem + 6 * 0.5rem + 6 * 0.125rem)) / 3);
//     }
//     #productImg {
//         height: 18.75rem;
//     }
// }

@include media-mixin ($ipad-breakpoint){
    .productCard {
        width:  calc((100% - (4 * 0.5rem + 4 * 0.5rem + 4 * 0.125rem)) / 2);
         /* 2cards=4spaces
        (100% - (4 * margin  + 4 * padding + 4 * border)) / number of cards per row */
    }
    #productImg {
        height: 20rem;
    }
}

// @media (max-width: 48rem) {         /*ipad: 768*/
//     .productCard {
//         width:  calc((100% - (4 * 0.5rem + 4 * 0.5rem + 4 * 0.125rem)) / 2);
//         /* 2cards=4spaces
//        (100% - (4 * margin  + 4 * padding + 4 * border)) / number of cards per row */
//     }
//     #productImg {
//         height: 20rem;
//     }
// }

@include media-mixin ($mobile-breakpoint){
    .productCard {
         width: 100%;
    }
}

// @media (max-width: 30rem) {         /*mobile-screen: pixel2, iphone  678 plus*/
//     .productCard {
//         width: 100%;
//     }
// }
