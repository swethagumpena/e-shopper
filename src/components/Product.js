/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ProductQuantity from './Counter';
import './Product.css';

class Product extends Component {
  render() {
    const { product, onIncrement, onDecrement } = this.props;
    return (
      <div className="product">
        <img id="product-img" src={product.url} alt="product" />
        <div className="container">
          <h4>{product.productName}</h4>
          <p>1 kg</p>
          <p>
            Rs
            {' '}
            {product.price}
          </p>
          <ProductQuantity
            value={product.quantity}
            onIncrement={onIncrement}
            onDecrement={onDecrement}
          />
        </div>
      </div>
    );
  }
}

// function Product() {
//   const { product } = this.props;
//   return (
//     <div className="Product">
//       <img id="product-img" src={product.src} alt="product" />
//       <div className="container">
//         <h4>{product.productName}</h4>
//         <p>1 kg</p>
//         <ProductQuantity
//           quantity={product.quantityInCart}
//         />
//       </div>
//     </div>
//   );
// }

// Product.propTypes = {
//   url: PropTypes.string,
//   productName: PropTypes.string.isRequired,
//   quantityInCart: PropTypes.number.isRequired,
// };

// Product.defaultProps = {
//   url: '',
// };

Product.propTypes = {
  product: PropTypes.objectOf.isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default Product;
