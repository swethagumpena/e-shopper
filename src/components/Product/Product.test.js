/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import Product from './Product';

describe('example', () => {
  test('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
});

describe(Product.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const mockProps = {
    product: {

      id: 1,
      name: 'Banana',
      quantity: 0,
      price: 40,
      url: 'xyz.com',

    },
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };

  test('should match snapshot', () => {
    const { container } = render(<Product {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  test('should display product content', () => {
    render(<Product {...mockProps} />);
    //   product={
    //     {
    //       id: 1,
    //       name: 'Banana',
    //       quantity: 0,
    //       price: 40,
    //       url: 'xyz.com',
    //     }
    //   }
    //   onIncrement={() => {}}
    //   onDecrement={() => {}}

    expect(screen.getByText('1 kg'));
    expect(screen.getByText('Banana'));
    expect(screen.getByText('MRP 40/-'));
    expect(screen.getByTestId('product-img'));
    expect(screen.getByTestId('counter'));
  });
});
