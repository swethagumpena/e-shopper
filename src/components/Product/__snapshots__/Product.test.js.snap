// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should match snapshot 1`] = `
<div>
  <div
    class="productCard"
  >
    <img
      alt="product"
      data-testid="product-img"
      id="productImg"
      src="xyz.com"
    />
    <div
      class="priceQuantityContainer"
    >
      <h4>
        Banana
      </h4>
      <p>
        1 kg
      </p>
      <div
        class="priceQuantity"
      >
        <p
          class="productPrice"
        >
          MRP 40/-
        </p>
        <div
          class="counter"
          data-testid="counter"
        >
          <button
            type="button"
          >
            -
          </button>
          <p
            class="containerQuantity0"
          >
            0 in basket
          </p>
          <button
            type="button"
          >
            +
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
